server/src/shared/infra/cache/redis-repository.ts

```diff
@@ -0,0 +1,64 @@
+import { RedisClient } from "../../../setup/cache/redis"
+import { AppError } from "../../core/app-error"
+import { Result } from "../../core/result"
+import { RedisEntity } from "./redis-entity"
+
+export type RedisSaveEntityResponse = Result<RedisSaveEntitySuccess, RedisSaveEntityError>
+export type RedisSaveEntitySuccess = boolean
+export type RedisSaveEntityError = AppError.UnexpectedError
+
+export type RedisGetEntityResponse<RedisEntityType> = Result<RedisEntityType, RedisGetEntityError>
+export type RedisGetEntityError = AppError.UnexpectedError
+
+export type RedisDeleteEntityResponse = Result<RedisDeleteEntitySuccess, RedisDeleteEntityError>
+export type RedisDeleteEntitySuccess = boolean
+export type RedisDeleteEntityError = AppError.UnexpectedError
+export interface RedisSetOptions {
+    mode: string
+    value: number
+}
+
+export class RedisRepository<RedisEntityType> {
+
+    async getEntity(entityKey: string): Promise<RedisGetEntityResponse<RedisEntityType>>{
+        return new Promise((resolve) => {
+            RedisClient().get(entityKey, (err, value) => {
+                if(err || value === null){
```

Comment:
I sound like a broken record, but you might want to run Prettier

Reply:
Finally configured it correctly - expect a very large formatting commit.